### ISSUE TITLE
[BUG] Support additional Elasticsearch types in pipeline validation

### ISSUE THREAD

**Issue by andrewstucki** [elastic/elastic-package#427]
Found at least one missing field type validation that's causing problems with one of our packages.

https://github.com/elastic/elastic-package/blob/17d8f26140461ed95b7cb37e6d6e001ca6357433/internal/fields/validate.go#L280

appears to be missing `boolean` validation logic, so we're missing test cases such as [this](https://github.com/elastic/integrations/blob/49795ac2dbc509b923cfd66cd8ab1a04553209a9/packages/auditd/data_stream/log/_dev/test/pipeline/test-auditd-raw.log-expected.json#L72) where a string "yes" isn't being properly mapped to a boolean value at ingest and is causing our package to be broken.

I think, at minimum, we should probably audit all of the Elasticsearch types we're likely to be using and make sure they're supported in our validation routines.

**Comment by jen-huang** [comment]
@mrodm do you know if this is still an issue?

**Comment by mrodm** [comment]
> @mrodm do you know if this is still an issue?

Yes, this issue is still valid @jen-huang . There is no validation for boolean types currently:
https://github.com/elastic/elastic-package/blob/a75fd6f3b9d77094c39f2dd72b54eae3561f55a5/internal/fields/validate.go#L715

**Comment by kpollich** [comment]
We haven't revisited this issue in a while, so we're closing it due to inactivity. Please reopen with a comment if you think this is a mistake. Thanks!
